portal-impl/src/com/liferay/portal/service/impl/GroupLocalServiceImpl.java

```diff
@@ -861,18 +861,16 @@ public Group deleteGroup(Group group) throws PortalException {
 					group.getGroupId(), RoleConstants.TYPE_SITE);
 			}
 			else {
-				if (!group.isStagingGroup() || group.isStagedRemotely()) {
```

Comment:
@JorgeFerrer @topolik: after looking at this a bit more I think that this is the right fix.  While I don't know for sure that Staging groups would never make use of an Owner role, I don't think we should have a situation where a group can be deleted without having its UserGroupRoles removed, irrespective of whether it's _supposed_ to have them or not. 

It seems much safer to me to do a blanket removal of UserGroupRoles, especially since we're deleting the Group anyway, so there's no chance we would need them in the future.

I would much rather do this since it prevents future regressions of this type.  It looks to me like we have been adding the Owner UserGroupRoles to Staging sites for quite a while, so changing that seems a lot more likely to cause regressions IMO.